### PR TITLE
Upgrade Elasticsearch from version 6.5.4 to 7.9.3 in docker-compose.yml

### PR DESCRIPTION
**Pull request by busma13**
Update docker-compose.yml file:

    Upgrade Elasticsearch from version 6.5.4 to version 7.9.3

refs:
#3392

### CODE REVIEW

**Review by godber — CHANGES_REQUESTED** [review]
Go ahead and follow the instructions on `upgrading` from 2 to 3 outlined here:

https://docs.docker.com/compose/compose-file/compose-versioning/#upgrading

There are only a few things and even fewer of them apply to our file.

Also, when you create a PR, you should add `refs: #NNNN` to the comment where `NNNN` is the issue number it addresses, it will link the PR and the Issue.

**Comment by busma13** [comment]
Using the docker-compose.yml file from the master branch, I cannot successfully build a teraslice docker image.  It looks like the node-base:14.21.3 image is amd64 and not compatible with the arm64 architecture of my MacBook M2.

```
peterluitjens@Peters-MacBook-Pro teraslice % docker compose up -d --build
[+] Building 0.7s (5/5) FINISHED                                                                                                                              docker:desktop-linux
 => [teraslice-master internal] load build definition from Dockerfile                                                                                                         0.0s
 => => transferring dockerfile: 991B                                                                                                                                          0.0s
 => [teraslice-master internal] load .dockerignore                                                                                                                            0.0s
 => => transferring context: 373B                                                                                                                                             0.0s
 => ERROR [teraslice-worker internal] load metadata for docker.io/terascope/node-base:14.21.3                                                                                 0.6s
 => [teraslice-worker internal] load .dockerignore                                                                                                                            0.0s
 => => transferring context: 373B                                                                                                                                             0.0s
 => [teraslice-worker internal] load build definition from Dockerfile                                                                                                         0.0s
 => => transferring dockerfile: 991B                                                                                                                                          0.0s
------
 > [teraslice-worker internal] load metadata for docker.io/terascope/node-base:14.21.3:
------
failed to solve: terascope/node-base:14.21.3: no match for platform in manifest sha256:0869240b6214519e5ab41188d1fd9cd257822940d89ab8c1c535c506f5b628b2: not found
```

**Comment by busma13** [comment]
I was able to build multiple docker images with different architectures following these instructions:

https://docs.docker.com/build/building/multi-platform/#getting-started

This stack overflow question gives some other workarounds for architecture mismatches during docker builds:

https://stackoverflow.com/questions/65612411/forcing-docker-to-use-linux-amd64-platform-by-default-on-macos

File: docker-compose.yml [hunk 1]
```diff
@@ -74,6 +74,7 @@ services:
       - IPC_LOCK
   kafka:
     image: terascope/kafka-zookeeper:v1.1.0
+    platform: linux/amd64
```
Comment (godber) [inline docker-compose.yml:77]:
```suggestion
    platform: linux/amd64  # platform specified to force amd64 on arm MacOS docker
```

**Review by godber — CHANGES_REQUESTED** [review]

**Comment by godber** [comment]
After running `docker compose up -d --build` on my Intel Mac, I got all of these images starting OK:

```
 docker compose ps
NAME                           IMAGE                                                 COMMAND                  SERVICE             CREATED             STATUS                    PORTS
teraslice-elasticsearch-1      docker.elastic.co/elasticsearch/elasticsearch:7.9.3   "/tini -- /usr/local…"   elasticsearch       53 seconds ago      Up 51 seconds (healthy)   0.0.0.0:9200->9200/tcp, 0.0.0.0:9300->9300/tcp
teraslice-kafka-1              terascope/kafka-zookeeper:v1.1.0                      "supervisord -n"         kafka               53 seconds ago      Up 52 seconds             0.0.0.0:2181->2181/tcp, 0.0.0.0:9092->9092/tcp
teraslice-teraslice-master-1   teraslice-teraslice-master                            "/sbin/tini -- node …"   teraslice-master    53 seconds ago      Up 52 seconds (healthy)   0.0.0.0:5678->5678/tcp
teraslice-teraslice-worker-1   teraslice-teraslice-worker                            "/sbin/tini -- node …"   teraslice-worker    53 seconds ago      Up 51 seconds (healthy)   5678/tcp
```

**Review by godber — APPROVED** [review]

**Pull request merged**